packages/@lwc/engine-core/src/framework/hot-swaps.ts

```diff
@@ -12,21 +12,37 @@ import { LightningElementConstructor } from './base-lightning-element';
 import { Template } from './template';
 import { markComponentAsDirty } from './component';
 import { isTemplateRegistered } from './secure-template';
-import { StylesheetFactory } from './stylesheet';
+import { StylesheetFactory, unrenderStylesheet } from './stylesheet';
 import { assertNotProd, flattenStylesheets } from './utils';
 import { WeakMultiMap } from './weak-multimap';
 
-const swappedTemplateMap = new WeakMap<Template, Template>();
-const swappedComponentMap = new WeakMap<LightningElementConstructor, LightningElementConstructor>();
-const swappedStyleMap = new WeakMap<StylesheetFactory, StylesheetFactory>();
+let swappedTemplateMap: WeakMap<Template, Template> = /*@__PURE__@*/ new WeakMap();
+let swappedComponentMap: WeakMap<LightningElementConstructor, LightningElementConstructor> =
+    /*@__PURE__@*/ new WeakMap();
+let swappedStyleMap: WeakMap<StylesheetFactory, StylesheetFactory> = /*@__PURE__@*/ new WeakMap();
 
 // The important thing here is the weak values – VMs are transient (one per component instance) and should be GC'ed,
 // so we don't want to create strong references to them.
 // The weak keys are kind of useless, because Templates, LightningElementConstructors, and StylesheetFactories are
 // never GC'ed. But maybe they will be someday, so we may as well use weak keys too.
-const activeTemplates: WeakMultiMap<Template, VM> = new WeakMultiMap();
-const activeComponents: WeakMultiMap<LightningElementConstructor, VM> = new WeakMultiMap();
-const activeStyles: WeakMultiMap<StylesheetFactory, VM> = new WeakMultiMap();
+// The "pure" annotations are so that Rollup knows for sure it can remove these from prod mode
+let activeTemplates: WeakMultiMap<Template, VM> = /*@__PURE__@*/ new WeakMultiMap();
+let activeComponents: WeakMultiMap<LightningElementConstructor, VM> =
+    /*@__PURE__@*/ new WeakMultiMap();
+let activeStyles: WeakMultiMap<StylesheetFactory, VM> = /*@__PURE__@*/ new WeakMultiMap();
```

Comment:
Nothing really changed here; I just added `PURE` annotations because I'm paranoid. I don't think this actually makes a difference, but I prefer to be safe in case bundlers have different heuristics (Rollup vs Webpack vs ESBuild, etc.).